src/controllers/controller.bar.js

```diff
@@ -95,6 +95,92 @@ defaults._set('horizontalBar', {
 	}
 });
 
+/**
+ * Computes the "optimal" sample size to maintain bars equally sized while preventing overlap.
+ * @private
+ */
+function computeMinSampleSize(scale, pixels) {
+	var min = scale.isHorizontal() ? scale.width : scale.height;
+	var ticks = scale.getTicks();
+	var prev, curr, i, ilen;
+
+	for (i = 0, ilen = pixels.length; i < ilen; ++i) {
+		min = i > 0 ? Math.min(min, pixels[i] - pixels[i - 1]) : min;
+	}
+
+	for (i = 0, ilen = ticks.length; i < ilen; ++i) {
+		curr = scale.getPixelForTick(i);
+		min = i > 0 ? Math.min(min, curr - prev) : min;
+		prev = curr;
+	}
+
+	return min;
+}
+
+/**
+ * Computes the "ideal" sample range based on the absolute bar thickness or, if undefined or
+ * null, uses the smallest interval (see computeMinSampleSize) that prevents bar overlapping.
+ * @private
+ */
+function computeFitSampleRange(index, ruler, options) {
+	var thickness = options.barThickness;
+	var count = ruler.stackCount;
+	var pixels = ruler.pixels;
```

Comment:
You could probably inline `pixels` and `curr` since they are only used once. I think that should actually result in smaller minified code too